### ISSUE TITLE
`AccountQuery` API returns all available/visible accounts if `android.permission.GET_ACCOUNTS` is NOT explicitly granted, regardless of values passed to `withTypes` 

### ISSUE THREAD

**Issue by discountbank** [vestrel00/contacts-android#338]
Testing on Device Android 28 with 23 accounts, 6 are gmail accounts.
Java
core lib version: latest
**Issue:**
Accounts API ignores filter by account types(regardless what I input) and **always returns EVERY accounts available(i.e 23)**
When there're several accounts with sync and sharing thousands of contacts querying all of them is not optimal, also buggy.
Code:
```
val accounts = Contacts(context).accounts().query()
    .withTypes("com.google")
    .find()

```
**Expected result = 6 gmail accounts**
**Actual result =  23 accounts(ALL)**

**Comment by vestrel00** [comment]
Hey! I'll take a look at this. 

Besides gmail, what other accounts do you have that contain contacts?

**Comment by vestrel00** [comment]
I agree that the account type doesn't matter. The reason why I'm asking for other account types from you aside from just "com.google" is that I also want to create other accounts that can sync contact for myself other than just Google. I've just been mainly using Google accounts for testing so I'm trying to expand my testing reach 😁

------

Anyways, in the meantime, you can workaround this issue by not using the `AccountsQuery` API provided in this library. Instead, use the Android `AccountManager` directly and see if that works for you. This way, you can continue your work while I investigate this issue. 

Please let me know if you are or are not seeing the same issue when using the `AccountManager` 🙏

https://developer.android.com/reference/android/accounts/AccountManager#getAccountsByType(java.lang.String)

**Comment by vestrel00** [comment]
I'm able to reproduce this issue using my physical Samsung Galaxy A71 API 33 !

I'll work on a fix!

**Comment by vestrel00** [comment]
Oh lol. I found the issue. I am stupid XD hahaha Look...

```kotlin
val visibleAccounts: MutableSet<Account> = accountManager.accounts.toMutableSet()

return if (
    cancel()
    || !accountsPermissions.canQueryAccounts()
    // No (visible) accounts in the system. No point in processing the rest of the query.
    || visibleAccounts.isEmpty()
) {
    AccountsQueryResult(visibleAccounts)
} else { ... }
```

The issue is that if `accountsPermissions.canQueryAccounts()` evaluates to false, meaning you don't have permission to query Accounts, all `visibleAccounts` is returned without filtering it 🤦 

What I also learned from this is that `GET_ACCOUNTS_PERMISSION` is not required to use https://developer.android.com/reference/android/accounts/AccountManager#getAccountsByType(java.lang.String) I'll keep that requirement for the API there anyways just to be safe.

-------

So, to "workaround" this issue without waiting for the next release, either...

(A) Use the Android `AccountManager` directly instead of the `AccountsQuery` API provided by this library.
(B) Make sure that you have `GET_ACCOUNTS_PERMISSION` when using the `AccountsQuery` API.

The fix for this will be included in the next release; [0.3.2](https://github.com/vestrel00/contacts-android/discussions/316)

**Comment by vestrel00** [comment]
This is included in release [0.3.2](https://github.com/vestrel00/contacts-android/releases/tag/0.3.2).
